fastlane/lib/fastlane/plugins/plugin_info_collector.rb

```diff
@@ -0,0 +1,67 @@
+module Fastlane
+  class PluginInfoCollector
+    def initialize(ui = PluginGeneratorUI.new)
+      @ui = ui
+    end
+
+    def collect_info
+      plugin_name = collect_plugin_name
+      author = collect_author
+
+      PluginInfo.new(plugin_name, author)
+    end
+
+    #
+    # Plugin name
+    #
+
+    def collect_plugin_name
+      plugin_name = nil
+      loop do
+        plugin_name = @ui.input("What would you like to be the name of your plugin?")
+        unless plugin_name_valid?(plugin_name)
+          fixed_name = fix_plugin_name(plugin_name)
+          if plugin_name_valid?(fixed_name)
+            plugin_name = fixed_name if @ui.confirm("Is '#{fixed_name}' OK?")
```

Comment:
'okay' ✅ 

The `PluginGeneratorUI` class is delegating to the `FastlaneCore` UI layer, so we will get that prompt behavior. 👍 We'll think some more about how much it is actually buying us.